### PR TITLE
This PR improves `exec`ing to the pod trhough CITF

### PR DESCRIPTION
**Pull request by a4abhishek**
improves `exec`ing to the pod trhough CITF.
- generalizes to handle multiple container pods
- empty string can be passed for `namespace` param if namespace is default
- empty string can be passed for `containerName` param in `ExecToPodThroughAPI` method, If the pod has only one container then it will work fine, otherwise it will through error.
- empty string can be passed for `containerName` param in `ExecToPodThroughKubectl` method, In this case it will be handled by `kubectl`.
- empty string can be passed for `containerName` param in `ExecToPod` method, In this case if there is only one container `ExecToPodThroughKubectl` will give the output, else it will be handled by `ExecToPodThroughKubectl` hence by `kubectl` eventually.
    
    Signed-off-by: Abhishek Kashyap <abhishek.kashyap@mayadata.io>

### CODE REVIEW

**Comment by Akash4927** [comment]
@a4abhishek Can you please resolve the conflicts?

**Comment by a4abhishek** [comment]
Sure @Akash4927

File: utils/k8s/k8s.go [hunk 1]
```diff
@@ -259,15 +259,20 @@ func (k8s K8S) YAMLApply(yamlPath string) error {
 	return nil
 }
 
-// ExecToPodThroughAPI uninterractively exec to the pod with the command specified.
```
Comment (sagarkrsd) [inline utils/k8s/k8s.go]: I am not sure about this word 'uninterractively', should be something like 'non-interactive'

File: utils/k8s/k8s.go [hunk 2]
```diff
@@ -259,15 +259,20 @@ func (k8s K8S) YAMLApply(yamlPath string) error {
...
+// ExecToPodThroughAPI uninterractively exec to the pod with the command specified with the help of client-go
```
Comment (sagarkrsd) [inline utils/k8s/k8s.go]: same here !!

File: utils/k8s/k8s.go [hunk 3]
```diff
@@ -259,15 +259,20 @@ func (k8s K8S) YAMLApply(yamlPath string) error {
...
 // :param string command: list of the str which specify the command.
+// :param string containerName: name of the container in the Pod. (If the Pod has only one container, then it can be Empty String)
 // :param string pod_name: Pod name
-// :param string namespace: namespace of the Pod.
+// :param string namespace: namespace of the Pod. (If it is blank string then, namespace will be default i.e. k8s.io/api/core/v1.NamespaceDefault)
 // :param io.Reader stdin: Standerd Input if necessary, otherwise `nil`
```
Comment (sagarkrsd) [inline utils/k8s/k8s.go]: Standerd should be standard !!

File: utils/k8s/k8s.go [hunk 4]
```diff
@@ -310,22 +320,52 @@ func (k8s K8S) ExecToPodThroughAPI(command, podName, namespace string, stdin io.
 	return stdout.String(), stderr.String(), nil
 }
 
+// ExecToPodThroughKubectl uninterractively exec to the pod with the command specified using `kubectl exec`
```
Comment (sagarkrsd) [inline utils/k8s/k8s.go]: same here !!

Comment (sagarkrsd) [reply utils/k8s/k8s.go]: One more comment @a4abhishek , Please correct the word trhough to through in the commit message !!

File: utils/k8s/k8s.go (hunk 1 above)
Comment (sagarkrsd) [reply utils/k8s/k8s.go]: You can write something like "ExecToPodThroughAPI performs non-interactive exec to the pod with the specified command".

**Comment by a4abhishek** [comment]
Addressed all your comments @sagarkrsd , have a look again.

File: utils/k8s/k8s.go [hunk 5]
```diff
@@ -310,22 +320,52 @@ func (k8s K8S) ExecToPodThroughAPI(command, podName, namespace string, stdin io.
...
+
+	// adding namespace if namespace is not blank string
+	if len(namespace) != 0 {
+		kubectlCommand += " -n " + namespace
+	}
+
+	// adding podName
+	kubectlCommand += " exec " + podName
+
+	// adding container name if contanerName is not a blank string
```
Comment (sagarkrsd) [inline utils/k8s/k8s.go]: contanerName => containerName

**Pull request merged**